src/reactjs_jsx_ppx_2.ml

```diff
@@ -0,0 +1,223 @@
+                    bsConfigLabel,
+                    PStr [{configRecord with pstr_desc = Pstr_eval ({innerConfigRecord with pexp_desc = Pexp_record (fields, b)}, a)}]
+                  )
+                }::restOfStructure)
+              end
+          | (_, recordFieldsWithoutJsx) -> raise (Invalid_argument "JSX: the file-level bs.config's {jsx: ...} config accepts only a version number")
+        end
+      | _ -> default_mapper.structure mapper structure
+    ) in
+
+  let handleJsxCall mapper callExpression callArguments attrs =
+    (match callExpression.pexp_desc with
+     | Pexp_ident caller ->
+       (match caller with
+        | {txt = Lident "createElement"} ->
+          raise (Invalid_argument "JSX: `createElement` should be preceeded by a module name.")
+        (* Foo.createElement prop1::foo prop2:bar children::[] () *)
+        (* no change *)
+        | {loc; txt = Ldot (modulePath, "createElement")} ->
+          let f = match !useNewJsxBehavior with
+            | Some 1 -> oldJSX
+            | Some 2 -> newJSX modulePath
+            | Some _ -> assert false
+            | None -> newJSX modulePath
+          in f mapper loc attrs callExpression callArguments
+        (* div prop1::foo prop2:bar children::[bla] () *)
+        (* turn that into ReactDOMRe.createElement props::(ReactDOMRe.props props1::foo props2::bar ()) [|bla|] *)
+        | {loc; txt = Lident id} ->
+          lowercaseCaller mapper loc attrs callArguments id
+        | {txt = Ldot (_, anythingNotCreateElement)} ->
+          raise (
+            Invalid_argument
+              ("JSX: the JSX attribute should be attached to a `YourModuleName.createElement` call. We saw `"
+               ^ anythingNotCreateElement
+               ^ "` instead"
+              )
+          )
+        | {txt = Lapply _} ->
+          (* don't think there's ever a case where this is reached *)
+          raise (
+            Invalid_argument "JSX: encountered a weird case while processing the code. Please report this!"
+          )
+       )
+     | anythingElseThanIdent ->
+       raise (
+         Invalid_argument "JSX: `createElement` should be preceeded by a simple, direct module name."
+       )
+    ) in
+
+  let expr =
+    (fun mapper expression -> match expression with
+       (* Function application with the @JSX attribute? *)
+       |
+         {
+           pexp_desc = Pexp_apply (callExpression, callArguments);
+           pexp_attributes
+         } when Syntax_util.attribute_exists "JSX" pexp_attributes ->
+         let attributesNoJsx =
+           List.filter (fun (attribute, _) -> attribute.txt <> "JSX") pexp_attributes in
+         handleJsxCall mapper callExpression callArguments attributesNoJsx
+       (* Delegate to the default mapper, a deep identity traversal *)
+       | e ->
+         default_mapper.expr mapper e) in
+
+  Reason_toolchain.To_current.copy_mapper { default_mapper with structure; expr }
```

Comment:
seems you can create a dedicated module for `To_current` to avoid pulling in such dependencies